src/main.rs

```diff
@@ -166,15 +219,41 @@ fn main() -> Result<()> {
             .await
             {
                 Ok(fetched_policy) => {
+                    if verify_enabled {
+                        if verified_manifest_digest.is_none() {
+                            // when deserializing keys we check that have keys to
+                            // verify. We will always have a digest manifest
+                            unreachable!();
+                        }
+                        verifier
+                            .verify_local_file_checksum(
+                                &fetched_policy.uri,
+                                docker_config.clone(),
+                                verified_manifest_digest.as_ref().unwrap(),
+                            )
+                            .await
+                            .map_err(|e| fatal_error(e.to_string()))
+                            .unwrap();
+                        info!(
+                            name = name.as_str(),
+                            sha256sum = verified_manifest_digest
+                                .as_ref()
+                                .unwrap_or(&"unknown".to_string())
+                                .as_str(),
+                            status = "verified-local-checksum",
+                            "policy download",
+                        );
+                    }
+
                     if let Ok(Some(policy_metadata)) =
                         Metadata::from_path(&fetched_policy.local_path)
                     {
                         info!(
                             name = name.as_str(),
                             path = fetched_policy.local_path.clone().into_os_string().to_str(),
-                            sha256sum = fetched_policy
-                                .digest()
-                                .unwrap_or_else(|_| "unknown".to_string())
+                            sha256sum = verified_manifest_digest
```

Comment:
Just realized that if we aren't verifying, we will not have the digest for logging here :facepalm: .

Reply:
Fixed on a rebase.